src/ql/pass/map/qubits/map/detail/free_cycle.h

```diff
@@ -21,151 +16,124 @@ namespace map {
 namespace detail {
 
 /**
- * FreeCycle: map each real qubit to the first cycle that it is free for use.
- *
- * In scheduling gates, qubit dependencies cause latencies. For each real qubit,
- * the first cycle that it is free to use is the cycle that the last gate that
- * was scheduled in the qubit, has just finished (i.e. in the previous cycle);
- * the map serves as a summary to ease scheduling next gates.
- *
- * Likewise, while mapping, swaps are scheduled just before a non-NN two-qubit
- * gate. Moreover, such swaps may involve real qubits on the path between the
- * real operand qubits of the gate, which may be different from the real operand
- * qubits. The evaluation of which path of swaps is best is, among other data,
- * based on which path causes the latency of the whole circuit to be extended
- * the least. This latency extension is measured from the data in the FreeCycle
- * map; so a FreeCycle map is part of each path of swaps that is evaluated for a
- * particular non-NN 2-qubit gate next to a FreeCycle map that is part of the
- * output stream (the main past).
+ * FreeCycle: map each reference to real qubit to the first cycle that it is free for use.
  *
- * Since gate durations are in nanoseconds, and one cycle is some fixed number
- * of nanoseconds, the duration is converted to a rounded-up number of cycles
- * when computing the added latency.
+ * This allows to compute, for each routing alternative, what the cost in terms of circuit depth
+ * will be - the goal of the minextend/minextendrc options in this router being to minimize the overall circuit depth.
+ * 
+ * This is also used in the options base and baserc to check whether the operands of a swap should be swapped
```

Comment:
So: base only counts swaps as metric; minextend counts circuit extension cycles.
Base doesn't need scheduling since it ignores cycles; so also doesn't need FreeCycle.
Minextend would need scheduling to find cycles but that is replaced by the use of FreeCycle.
The purpose of interchanging swap operands is to decrease the cycle extension; the swap always counts as one swap in base.